### ISSUE TITLE
Problems connecting to the VPN

### ISSUE THREAD

**Issue by ernlavr** [diku-dk/dpp-e2022-pub#1]
Hello,

I'm following the VPN setup guide but I'm having issues with the setup, I'm using Ubuntu 22.04.

### _**OpenConnect**_
I tried using the `openconnect` package and running the `$ sudo openconnect vpn.ku.dk` command although I'm getting refused access (I tried reconnecting a bunch of times and really cross-checking that I'm entering my usual password that I use to access KUNet). What could be the cause of this?

![image](https://user-images.githubusercontent.com/40242518/202857463-fd5cf49a-9d05-4062-b04b-6db8a3c70db8.png)

### _**Legacy VPN**_
Afterwards I've resorted to giving the "legacy" connection an attempt. Everything went fine until I was missing "Legacy Proposals" in the IPSec settings, see the screenshot below for what I have. Should I check any of the unchecked boxes here? Besides the "Legacy Proposals", I was able to set all other settings as per the documentation.
![image](https://user-images.githubusercontent.com/40242518/202857605-f3aa39b7-c870-4063-82f6-bb892ceb034f.png)

**Comment by zfnmxt** [comment]
Presuming this is to get access to the GPU machines---have you tried just SSHing into the gateway server? You shouldn't need a VPN for that.

**Comment by athas** [comment]
Do you have KU 2FA setup?  I just tried the instructions myself and it works for me.  However, I had to perform 2FA authentication via my phone.

**Comment by Munksgaard** [comment]
Do you have MFA setup for your KU-account? I believe that is necessary to use the new VPN.

**Comment by athas** [comment]
Maybe students don't have access to the new VPN?

**Comment by Munksgaard** [comment]
http://mfa.ku.dk/ For setting up 2FA/MFA

**Comment by ernlavr** [comment]
> Presuming this is to get access to the GPU machines---have you tried just SSHing into the gateway server? You shouldn't need a VPN for that.

Yes I have and the SSH endpoint does not authenticate me with my KU password here is my config
EDIT: I just noticed that my hostname is ending with `.domain` is that meant to be a placeholder that I was supposed to swap out with an actual domain, i.e. `.dk` or `.ku.dk`?
![image](https://user-images.githubusercontent.com/40242518/202860698-215d8a54-bd00-4116-a1d4-ba1abdcea2ab.png)

**Comment by ernlavr** [comment]
> Do you have MFA setup for your KU-account? I believe that is necessary to use the new VPN.

Yes, I normally use NemID as my 2FA to access KUNet

**Comment by athas** [comment]
For some reason you had not been added to the Futhark server access group.  Try again.  The `.domain` ending is correct; it's a local domain name.

**Comment by ernlavr** [comment]
> For some reason you had not been added to the Futhark server access group. Try again. The `.domain` ending is correct; it's a local domain name.

Hmmm.. I still can't seem to be able to SSH into the system although I received a confirmation email to my student inbox that changes to my identity have been processed. Should I perhaps wait until tomorrow to let the IT-backend update the issued changes?
![image](https://user-images.githubusercontent.com/40242518/202864281-723821ef-95c2-4c04-acbb-901c1636ffd9.png)

**Comment by FrederikBerthelsen** [comment]
I have the same problem as you, I get permission denied everytime I ssh. I am positively sure that I write my password correctly. I also have 2FA activated, by using my phone.
I read the guide as you can either connect to the VPN or you can go through the gateway server, is this correct? Or do I need to do both?

**Comment by athas** [comment]
Either should work.  I'm not sure why you are unable to login to the SSH gateway.

**Comment by ernlavr** [comment]
> Either should work. I'm not sure why you are unable to login to the SSH gateway.

But how should we proceed? Surely we need to get this stuff running one way or another.. I can't imagine using university local network being a sustainable solution. I need to be able to run this stuff from my home setup.. Shall I just talk to the TA about this tomorrow?

**Comment by athas** [comment]
There is no TA, but we'll work through it in the first lab tomorrow. I'll also contact KU-IT to figure out what is going on.

**Comment by athas** [comment]
Can you connect to the *old* VPN?

**Comment by athas** [comment]
Can you SSH to ssh-diku.science.ku.dk with your KU username?  (Note different domain than above.)

**Comment by FrederikBerthelsen** [comment]
That worked for me (SSH to ssh-diku.science.ku.dk). From there I could also SSH to the GPU machines.

**Comment by ernlavr** [comment]
> Can you connect to the _old_ VPN?

No, my Ubuntu VPN GUI is missing a checkbox for "Legacy Proposals" as required by the documentation, and as shown in the original post. Besides the missing checkbox I managed to fulfill all other steps in the documentation but my laptop still doesn't connect to the VPN

> That worked for me (SSH to ssh-diku.science.ku.dk). From there I could also SSH to the GPU machines.

Did you simply swap out the domain in your SSH config file from _username@ssh-diku-apl.science.ku.dk_ to _username@ssh-diku.science.ku.dk_ ? I tried this but unfortunately this does not work for me..

**Comment by athas** [comment]
Can you SSH directly to ssh-diku.science.ku.dk?

**Comment by ernlavr** [comment]
> ssh-diku.science.ku.dk

Doesn't seem so, I ran `$ ssh lfc161@ssh-diku.science.ku.dk` and still getting denied access
Also tried swapping the proxy jump entry in the SSH config without any luck.

**Comment by athas** [comment]
I received an answer from KU IT:

```
Hej Troels

For at de studerende kan komme på VPN skal de også have en gruppe som enten hedder
au-sci-diku--ansat eller au-sci-diku--associeret. Jeg kiggede på tre tilfældige med gruppen
srv-futharkhpa-users og fandt at de alle havde det. Men måske mangler nogen. 

Det næste trin er at de skal have MFA oprettet, det gøres via vpn.ku.dk eller et opkald til it
supporten.

Hvis en bruger ikke er medlem af en gruppe som giver VPN adgang skal de tilføjes til den.

når det kommer på plads, så kommer server adgangen helt af sig selv.

Der er oprettet en firewall regel som tillader trafik fra diku vpn til futharkhpa01-3 hvis man har
gruppe srv-futharkhpa-users
```

So apparently server access does not imply VPN access.  As an extra wrinkle, I do not have the rights to grant you VPN access.  My conversation with KU IT continues.  I estimate that this will be fixed before it becomes too urgent.

**Comment by athas** [comment]
If you use the app-based 2FA (with "NetIQ Advanced Authenticator"), then you appear to be able to connect to the VPN and directly ssh to e.g. `futharkhpa01fl.unicph.domain`.  I think KU IT is working on giving students access to the SSH bastion servers (which will not require VPN).
